test/prune-publish.test.js

```diff
@@ -78,10 +80,13 @@ test('publish many messages that might need pruning', (t) => {
         })
     )
 
-    //console.time('publish')
+    console.log('publishing', n)
+    console.time('publish')
     Promise.all(publishes)
       .then(async () => {
-        //console.timeEnd('publish')
+        console.timeEnd('publish')
+
+        t.pass('published all the messages')
```

Comment:
This was erroring in CI when n=5000

https://github.com/ssbc/ssb-tribes2/actions/runs/4386900747/jobs/7681532898

```
/home/runner/work/ssb-tribes2/ssb-tribes2/node_modules/.pnpm/jitdb@7.0.7_op2ybb55xfa424zffwdfaabewq/node_modules/jitdb/index.js:1278
          filterRecord(sliced[i].seq, filters, recBufferCache, done())
                                 ^

TypeError: Cannot read properties of undefined (reading 'seq')
    at getMoreResults (/home/runner/work/ssb-tribes2/ssb-tribes2/node_modules/.pnpm/jitdb@7.0.7_op2ybb55xfa424zffwdfaabewq/node_modules/jitdb/index.js:1278:34)
    at getMessagesFromBitsetSlice (/home/runner/work/ssb-tribes2/ssb-tribes2/node_modules/.pnpm/jitdb@7.0.7_op2ybb55xfa424zffwdfaabewq/node_modules/jitdb/index.js:1283:7)
    at /home/runner/work/ssb-tribes2/ssb-tribes2/node_modules/.pnpm/jitdb@7.0.7_op2ybb55xfa424zffwdfaabewq/node_modules/jitdb/index.js:1399:9
    at CollectStream._cb (/home/runner/work/ssb-tribes2/ssb-tribes2/node_modules/.pnpm/jitdb@7.0.7_op2ybb55xfa424zffwdfaabewq/node_modules/jitdb/index.js:1078:14)
    at CollectStream.end (/home/runner/work/ssb-tribes2/ssb-tribes2/node_modules/.pnpm/push-stream@11.2.0/node_modules/push-stream/sinks/collect.js:16:15)
    at AsyncMapStream.end (/home/runner/work/ssb-tribes2/ssb-tribes2/node_modules/.pnpm/push-stream@11.2.0/node_modules/push-stream/throughs/async-map.js:28:20)
    at AsyncMapStream.end (/home/runner/work/ssb-tribes2/ssb-tribes2/node_modules/.pnpm/push-stream@11.2.0/node_modules/push-stream/throughs/async-map.js:28:20)
    at AsyncMapStream.end (/home/runner/work/ssb-tribes2/ssb-tribes2/node_modules/.pnpm/push-stream@11.2.0/node_modules/push-stream/throughs/async-map.js:28:20)
    at ValueStream.resume (/home/runner/work/ssb-tribes2/ssb-tribes2/node_modules/.pnpm/push-stream@11.2.0/node_modules/push-stream/sources/values.js:19:51)
    at AsyncMapStream.resume (/home/runner/work/ssb-tribes2/ssb-tribes2/node_modules/.pnpm/push-stream@11.2.0/node_modules/push-stream/throughs/through.js:17:19)
```

Reply:
Did some timing with different `n`s

```
# publish many messages that might need pruning
publishing 5
publish: 4.873s
ok 3 published all the messages
```

```
# publish many messages that might need pruning
publishing 50
publish: 34.273s
ok 3 published all the messages
```

```
# publish many messages that might need pruning
publishing 100
publish: 1:01.850 (m:ss.mmm)
ok 3 published all the messages
```

Reply:
On master the timing is approx

```
publishing 5000
publish: 13.470s
```

Reply:
Haven't changed `publish` in this PR.

Diff between box2 5 and 6 https://github.com/ssbc/ssb-box2/compare/v5.0.0...v6.0.0

Diff between keyring 4.1.0 and 5.0.0 https://gitlab.com/ahau/lib/ssb-keyring/-/compare/v4.1.0...v5.0.0?from_project_id=28910207&straight=false